sycl/include/sycl/esimd/esimd_math.hpp

```diff
@@ -0,0 +1,1950 @@
+                              simd<ftype, SZ>>::type                           \
+          esimd_##name(U src0, simd<ftype, SZ> src1, int flag = GENX_NOSAT) {  \
+    simd<ftype, SZ> Src0 = src0;                                               \
+    return esimd_##name(Src0, src1, flag);                                     \
+  }                                                                            \
+  ESIMD_NODEBUG ESIMD_INLINE ftype esimd_##name(ftype src0, ftype src1,        \
+                                                int flag = GENX_NOSAT) {       \
+    simd<ftype, 1> Src0 = src0;                                                \
+    simd<ftype, 1> Src1 = src1;                                                \
+    simd<ftype, 1> Result = esimd_##name(Src0, Src1, flag);                    \
+    return Result[0];                                                          \
+  }
+
+ESIMD_INTRINSIC_DEF(float, pow)
+
+ESIMD_INTRINSIC_DEF(float, div_ieee)
+ESIMD_INTRINSIC_DEF(double, div_ieee)
+
+#undef ESIMD_INTRINSIC_DEF
+
+// esimd_sincos
+template <int SZ, typename U>
+ESIMD_NODEBUG ESIMD_INLINE simd<float, SZ>
+esimd_sincos(simd<float, SZ> &dstcos, U src0, int flag = GENX_NOSAT) {
+  dstcos = esimd_cos(src0, flag);
+  return esimd_sin(src0, flag);
+}
+
+// esimd_atan
+
+#define ESIMD_HDR_CONST_PI 3.1415926535897932384626433832795
+
+template <typename T, int SZ>
+ESIMD_NODEBUG ESIMD_INLINE
+    typename std::enable_if<std::is_floating_point<T>::value, simd<T, SZ>>::type
+    esimd_atan(simd<T, SZ> src0, int flag = GENX_NOSAT) {
+  simd<T, SZ> Src0 = esimd_abs(src0);
+
+  simd<ushort, SZ> Neg = src0 < T(0.0);
+  simd<ushort, SZ> Gt1 = Src0 > T(1.0);
+
+  Src0.merge(esimd_inv(Src0), Gt1);
+
+  simd<T, SZ> Src0P2 = Src0 * Src0;
+  simd<T, SZ> Src0P4 = Src0P2 * Src0P2;
+
+  simd<T, SZ> Result = (Src0P4 * T(0.185696) +
+                        ((Src0 * T(0.787997) + T(0.63693)) * Src0P2) + Src0) /
+                       (((((Src0 * -T(0.000121387) + T(0.00202308)) * Src0P2) +
+                          (Src0 * -T(0.0149145)) + T(0.182569)) *
+                         Src0P4) +
+                        ((Src0 * T(0.395889) + T(1.12158)) * Src0P2) +
+                        (Src0 * T(0.636918)) + T(1.0));
+
+  Result.merge(Result - T(ESIMD_HDR_CONST_PI / 2.0), Gt1);
+  Result.merge(Result, Neg);
+
+  if (flag != GENX_SAT)
+    return Result;
+
+  return esimd_sat<T>(Result);
+}
+
+template <typename T>
+ESIMD_NODEBUG ESIMD_INLINE
+    typename std::enable_if<std::is_floating_point<T>::value, T>::type
+    esimd_atan(T src0, int flag = GENX_NOSAT) {
+  simd<T, 1> Src0 = src0;
+  simd<T, 1> Result = esimd_atan(Src0, flag);
+  return Result[0];
+}
+
+// esimd_acos
+
+template <typename T, int SZ>
+ESIMD_NODEBUG ESIMD_INLINE
+    typename std::enable_if<std::is_floating_point<T>::value, simd<T, SZ>>::type
+    esimd_acos(simd<T, SZ> src0, int flag = GENX_NOSAT) {
```

Comment:
`GENX_NOSAT` should be a `contexpr` variable?

Reply:
it is an enum defined in esimd_enum.hpp

Reply:
no change on this one